controllers/release_controller.go

```diff
@@ -45,103 +44,105 @@ type ReleaseReconciler struct {
-		err := r.Get(ctx, types.NamespacedName{
-			Namespace: r.OrganizationNamespace,
-			Name:      strategyName,
-		}, releaseStrategy)
+// getReleaseLink loads and returns the ReleaseLink referenced in the given Release.
+func (r *ReleaseReconciler) getReleaseLink(ctx context.Context, release *v1alpha1.Release) (*v1alpha1.ReleaseLink, error) {
+	releaseLink := &v1alpha1.ReleaseLink{}
+	err := r.Get(ctx, types.NamespacedName{
+		Namespace: release.Namespace,
+		Name:      release.Spec.ReleaseLink,
+	}, releaseLink)
 
-		if err == nil {
-			strategies = append(strategies, releaseStrategy)
-		}
+	if err != nil {
+		return nil, err
 	}
 
-	return strategies
+	return releaseLink, nil
 }
 
-// triggerReleases triggers a new release for each strategy referenced in the given component.
-func (r *ReleaseReconciler) triggerReleases(ctx context.Context, component *hasv1alpha1.Component) {
-	log := r.Log.WithValues("Component.Name", component.Name, "Component.Namespace", component.Namespace)
-
-	log.Info("Fetching release strategies from component")
-	if len(component.Spec.ReleaseStrategies) == 0 {
-		log.Info("Component has not declared any release strategy",
-			"Component.Spec.ReleaseStrategies", component.Spec.ReleaseStrategies)
-		return
-	}
+// getReleaseStrategy loads and returns the ReleaseStrategy referenced in the given ReleaseLink.
+func (r *ReleaseReconciler) getReleaseStrategy(ctx context.Context, releaseLink *v1alpha1.ReleaseLink) (*v1alpha1.ReleaseStrategy, error) {
+	releaseStrategy := &v1alpha1.ReleaseStrategy{}
+	err := r.Get(ctx, types.NamespacedName{
+		Namespace: releaseLink.Spec.Target,
+		Name:      releaseLink.Spec.ReleaseStrategy,
+	}, releaseStrategy)
 
-	strategies := r.getReleaseStrategies(ctx, component)
-	if len(strategies) == 0 {
-		log.Info("No release strategy found in the cluster",
-			"Component.Spec.ReleaseStrategies", component.Spec.ReleaseStrategies)
-		return
+	if err != nil {
+		return nil, err
 	}
 
-	for _, strategy := range strategies {
-		log.Info("Triggering release", "ReleaseStrategy", strategy.Name)
-		pipelineRun := tekton.CreatePipelineRunFromReleaseStrategy(strategy, r.OrganizationNamespace, component)
-		err := r.Create(ctx, pipelineRun)
-		if err != nil {
-			log.Error(err, "Unable to trigger a release", "ReleaseStrategy.Name", strategy.Name)
-		}
-		log.Info("Release triggered",
-			"PipelineRun.Name", pipelineRun.Name,
-			"PipelineRun.Namespace", pipelineRun.Namespace)
-	}
+	return releaseStrategy, nil
 }
 
 // SetupWithManager sets up the controller with the Manager which monitors PipelineRuns and filter them
 // using a predicate to filter out everything but just succeeded Build PipelineRuns.
 func (r *ReleaseReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&tektonv1beta1.PipelineRun{}).
-		WithEventFilter(tekton.BuildPipelineRunSucceededPredicate()).
+		For(&v1alpha1.Release{}).
+		WithEventFilter(predicate.GenerationChangedPredicate{}).
```

Comment:
what does this call? I guess something with the CreateFunc?

Reply:
This is to ignore status changes. This changes don’t modify the generation.